File: Windows/Leonino/HexParser/HexRecord.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hex
{
	public class HexRecord
	{
		public const int DataType = 0;
		public const int EOF = 1;
		public const int ExtendedSegmentAddress = 2;
		public const int ExtendedLinearAddressRecord = 4;


		public int Type { get; set; }
		public int ExtendedAddress { get; set; }
		public int Address { get; set; }
		public int Length { get; set; }
		public byte[] Data { get; set; }
		public int CheckSum { get; set; }
		public bool IsOk
		{
			get
			{
				//check content and checksum
				return true;
			}
		}

		public int FullAddress
		{
			get
			{
				return ExtendedAddress + Address;
			}
		}
	}
}

File: IDE/Windows/LeoninoUSBJava/src/hex/HexRecord.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package hex;

/**
 *
 * @author rudarobson
 */
public class HexRecord {

    public final static int DataType = 0;
    public final static int EOF = 1;
    public final static int ExtendedSegmentAddress = 2;
    public final static int ExtendedLinearAddressRecord = 4;
    
    private int type;
    private int extendedAddress;
    private int address;
    private int length;
    private byte[] data = null;
    private int checkSum;

    public int getFullAddress() {
        return extendedAddress + address;
    }
    
    public int getAddress() {
        return address;
    }

    public void setAddress(int Address) {
        this.address = Address;
    }

    public int getCheckSum() {
        return checkSum;
    }

    public void setCheckSum(int CheckSum) {
        this.checkSum = CheckSum;
    }

    public byte[] getData() {
        return data;
    }

    public void setData(byte[] Data) {
        this.data = Data;
    }

    public int getExtendedAddress() {
        return extendedAddress;
    }

    public void setExtendedAddress(int ExtendedAddress) {
        this.extendedAddress = ExtendedAddress;
    }

    public int getLength() {
        return length;
    }

    public void setLength(int Length) {
        this.length = Length;
    }

    public int getType() {
        return type;
    }

    public void setType(int Type) {
        this.type = Type;
    }
}
